crates/pathfinder/src/state/block_hash.rs

```diff
@@ -444,6 +432,38 @@ pub fn compute_final_hash(header: &BlockHeaderData) -> Result<BlockHash> {
     Ok(BlockHash(hasher.finish().into()))
 }
 
+fn compute_final_hash_new(header: &BlockHeaderData) -> Result<BlockHash> {
```

Comment:
This one would then be `compute_hash_v1`. 

_Bonus: Add a comment on what changes were introduced._